File: map_tracker_spring/src/main/java/com/isquibly/maptracker/region/RegionBucket.java
package com.isquibly.maptracker.region;

import com.isquibly.maptracker.dto.UserEntry;

import java.time.OffsetDateTime;
import java.util.List;

/**
 * Response bucket for a single named region.
 * Contains all active riders within the region's bounding box.
 */
public record RegionBucket(
        String regionId,
        String regionName,
        OffsetDateTime updatedAt,
        List<UserEntry> riders
) {}

File: map_tracker_flutter/lib/models/region_bucket.dart
import 'user_entry.dart';

class RegionBucket {
  final String regionId;
  final String regionName;
  final DateTime updatedAt;
  final List<UserEntry> riders;

  const RegionBucket({
    required this.regionId,
    required this.regionName,
    required this.updatedAt,
    required this.riders,
  });

  factory RegionBucket.fromJson(Map<String, dynamic> json) => RegionBucket(
        regionId:   json['region_id']   as String,
        regionName: json['region_name'] as String,
        updatedAt:  DateTime.parse(json['updated_at'] as String),
        riders: (json['riders'] as List<dynamic>)
            .map((e) => UserEntry.fromJson(e as Map<String, dynamic>))
            .toList(),
      );
}
